### PR TITLE
un-revert: #29555

### PR DESCRIPTION
**Pull request by efritz**
Reapplies #29555 with fixes.

image

Fixed some hidden panics causing CI to report mysterious (but true) failures.

### CODE REVIEW

**Comment by efritz** [comment]
[Dry run here](https://buildkite.com/sourcegraph/sourcegraph/builds/125341)

File: internal/database/migration/runner/testdata/query-error/10001_first.up.sql [hunk 1]
```diff
@@ -1 +1,5 @@
+-- +++
+-- parent: 10000
```
Comment (slimsag) [inline internal/database/migration/runner/testdata/query-error/10001_first.up.sql:2]: where can I learn what these annotations do? (I'm not up to speed with any of the migration stuff you've done yet)

**Review by slimsag — APPROVED** [review]
I see revert, I approve.

File: internal/database/migration/runner/testdata/query-error/10001_first.up.sql (hunk 1 above)
Comment (slimsag) [reply internal/database/migration/runner/testdata/query-error/10001_first.up.sql:2]: nvm https://github.com/sourcegraph/sourcegraph/pull/29555#issuecomment-1011075641 answered it.

Comment (efritz) [reply internal/database/migration/runner/testdata/query-error/10001_first.up.sql:2]: Currently it just lists the previous filename. This is a temporary state that will be better documented in the future (right now it should be automated when you create a new migration file).

**Pull request merged**